src/main/java/com/kumuluz/ee/config/microprofile/adapters/ConfigSourceAdapter.java

```diff
@@ -93,4 +104,35 @@ private void buildPropertiesMap(Map<String, String> map, String prefix) {
             }
         }
     }
+	
+	private String buildArray(String propertyName, int size) {
+		StringBuilder sb = new StringBuilder();
+
+		for (int i = 0; i < size; i++) {
+			String prefix = String.format("%s[%d]", propertyName, i);
+	        Optional<List<String>> objectKeys = this.configurationSource.getMapKeys(prefix);
+
+	        //array item is an object, so we just toString() it
+	        if (objectKeys.isPresent()) {
+	            Map<String, String> map = new HashMap<>();
```

Comment:
If array item is an object just skip it. As said previously the Config MP specification does not support key hierarchy. If users need this functionality they will be better off using our configuration framework until proper hirearchy gets added to Config MP specification.

Reply:
>>If array item is an object just skip it.

Won't this break the consistency of the array? I think we should at least maintain the expected size and put nulls or blanks on the corresponding indexes.

Reply:
No, if we process these object in any way they would be inconsistent with the configuration supplied.

Reply:
I don't get what you just said, if you put `sb.append("")` instead of the code block you quoted, no processing would happen whatsoever.

Reply:
Exactly. That's what I meant by skipping object.